packages/hyperion-flowlet/test/FlowletWrapper.test.ts

```diff
@@ -153,4 +153,36 @@ describe("test flow of flowlets", () => {
     const allSettled = Promise.allSettled([p0, p1, p2]);
     expect(getTriggerFlowlet(allSettled)?.getFullName()).toMatch(/Promise.allSettled\(\d+&\d+&\d+\)/);
   });
+
+  test("Trigger flowlet propagation", (done) => {
+    const main = flowletManager.top() ?? flowletManager.push(new Flowlet("main"));
+    const tf1 = new flowletManager.flowletCtor("tf1");
+    main.data.triggerFlowlet = tf1;
+
+    const div1 = document.createElement("div");
+    document.body.appendChild(div1);
+    const div2 = document.createElement("div");
+    div1.appendChild(div2);
+
+    const tf2 = new Flowlet("tf2");
+    tf2.data.triggerFlowlet = tf2;
+    div1.addEventListener('click', event => {
+      setTriggerFlowlet(event, tf2);
+    }, true);
+
+    div2.addEventListener("click", event => {
+      setTimeout(() => {
+        Promise.resolve().then(() => {
+          const top = flowletManager.top();
+          console.log("Top: ", top.getFullName());
+          const tf = top.data.triggerFlowlet;
+          console.log("TF: ", tf?.getFullName());
+          expect(tf).toStrictEqual(tf2);
```

Comment:
Curious, was the trigger flowlet value here different before the changes applied in this PR?

Reply:
I was hoping to create a situation like that, but it didn't happen and TF continued to work great. 
Although, after this test, I realized we probably want to intercept `dispatchEvent` api and link to the what ever cause this api call. 
That might help us distinguish between react triggered events and actual ones better. But I will add this functionality later separately.